src/backend/stream.rs

```diff
@@ -221,7 +243,23 @@ impl<'ctx> PulseStream<'ctx> {
                 return;
             }
 
-            if stm.input_stream.is_none() {
+            if stm.input_stream.is_some() {
+                cubeb_logv!("duplex output callback. input frame count");
+                let nsamples = nbytes / stm.output_sample_spec.sample_size();
```

Comment:
Can we always read the data pushed into the buffer? What is the relationship between `nsamples` and `read_samples` in `read_data`? I guess the `sample_size()` of `output_sample_spec` and `input_sample_spec` are same since they are validated in [cubeb.c](https://github.com/kinetiknz/cubeb/blob/0f799155fc1689c4932589b0240b604374419615/src/cubeb.c#L91). Thus, their relationship should depend on `nbytes` with `read_size`  in `read_data`, but I am not familiar with them.